server/data/data-library/src/test/java/org/apache/james/mailrepository/mock/MockMailRepositoryStore.java

```diff
@@ -46,6 +47,15 @@ public MailRepository select(MailRepositoryUrl url) {
         return Optional.ofNullable(storedObjectMap.get(url));
     }
 
+    @Override
+    public Stream<MailRepository> getByPath(MailRepositoryPath path) {
+        return storedObjectMap
+                .keySet()
+                .stream()
+                .filter((MailRepositoryUrl key) -> key.asString().substring(key.getProtocol().getValue().length() + 2).equals(path.asString()))
```

Comment:
We need a `MailRepositoryUrl::getPath` method ;-)

Reply:
I think it deserve  a`private final String path` & getter